### ISSUE TITLE
parse expression failed in v1.13.3  but not found in v1.13.2 

### ISSUE THREAD

**Issue by watertreestar** [camunda/feel-scala#362]
**Describe the bug**
When i use the same expression to parse by feel-engine v1.13.3, it not work expectlly with failures, While it not occur using feel-engine v1.13.2

**To Reproduce**
1. create a expression `(count(oneMonthBuddies)>0) and (some emp in oneMonthBuddies satisfies (is defined(emp.buddy.email)))`
2. prepare feel-repl.sc for amm. [feel get stated refrence](https://camunda.github.io/feel-scala/docs/reference/). I add a new function for test, as follows:
```scala
def test(expression: String): Any = {
  val result = feelEngine.parseExpression(expression)
  printResult(result)
}
```
3. start REPL by ` amm --predef feel-repl.sc`(the version of feel in feel-repl.sc is 1.13.3)
4. execute `test("(count(oneMonthBuddies)>0) and (some emp in oneMonthBuddies satisfies (is defined(emp.buddy.email)))")` to get parse result
5. parse result is: `Failure(failed to parse expression '(count(oneMonthBuddies)>0) and (some emp in oneMonthBuddies satisfies (is defined(emp.buddy.email)))': Expected (continuation | StringIn("<=", ">=", "<", ">", "!=", "=") | "between" | "in" | "and" | "or" | ".."):1:100, found ")")`
6. change version to 1.13.2 in feel-repl.sc is parse successfully

**Expected behavior**
Parse successfully in both version

**Environment**
* FEEL engine version: [1.13.2,1.13.3]
* Affects: 
  * Zeebe broker: [1.2.4]

**Comment by saig0** [comment]
@watertreestar thank you for reporting and providing a detailed description :+1: 

I'll check if the bug still exists in the upcoming version `1.4.0`. I did major changes in the parser.
Feel free to contribute a unit test :rocket:

**Comment by saig0** [comment]
The bug is already fixed in version `1.4.0`. I added a new test case for this boolean expression with #374.
